src/query/ee/src/table_lock/table_lock_manager.rs

```diff
@@ -66,12 +104,8 @@ impl TableLockHandler for RealTableLockHandler {
             }
 
             // Get the previous revision, watch the delete event.
-            let lock_key = TableLockKey {
-                table_id,
-                revision: reply[position - 1],
-            };
             let req = WatchRequest {
-                key: lock_key.to_string_key(),
+                key: lock.watch_delete_key(reply[position - 1]),
```

Comment:
Maybe dead lock when lock key is expired?

Reply:
https://github.com/datafuselabs/databend/blob/6fc5bb2783f40d4cd6cc9e8a30317c046ffba073/src/meta/service/tests/it/grpc/metasrv_grpc_watch.rs#L324 
Also can get the delete event when key is expired.